deltacat/compute/compactor/steps/materialize.py

```diff
@@ -185,23 +214,55 @@ def _materialize(compacted_tables: List[pa.Table]) -> MaterializeResult:
                 f" to download delta locator {delta_locator} with entry ID {src_file_idx_np.item()}"
                 f" is: {download_delta_manifest_entry_time}s"
             )
-            mask_pylist = list(repeat(False, len(pa_table)))
+            record_numbers_tpl_copy = copy.deepcopy(record_numbers_tpl)
+            record_numbers_length_list = chain.from_iterable(record_numbers_tpl_copy)
+            record_numbers_length = sum(1 for item in record_numbers_length_list)
             record_numbers = chain.from_iterable(record_numbers_tpl)
-            # TODO(raghumdani): reference the same file URIs while writing the files
-            # instead of copying the data over and creating new files.
-            for record_number in record_numbers:
-                mask_pylist[record_number] = True
-            mask = pa.array(mask_pylist)
-            pa_table = pa_table.filter(mask)
-            record_batch_tables.append(pa_table)
-            if record_batch_tables.has_batches():
-                batched_tables = record_batch_tables.evict()
-                materialized_results.append(_materialize(batched_tables))
+            if (
+                record_numbers_length == len(pa_table)
+                and src_file_partition_locator != source_partition_locator
+            ):
+                logger.info(
+                    f"Untouched manifest file found, "
+                    f"record numbers length: {record_numbers_length} "
+                    f"same as downloaded table length: {len(pa_table)}"
+                )
```

Comment:
We may be unnecessarily logging each file as there can be thousands of files in a single delta. However, it can be a debug log if you think it will be useful. Also, consider how it can impact performance.